packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -128,44 +132,65 @@ export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TCh
             return commitToRebase.changeset;
         }
 
+        let nextRevisionNum = -1;
         const changeRebasedToRef = branch.localChanges.reduceRight(
             (newChange, branchCommit) =>
                 this.changeFamily.rebaser.rebase(
                     newChange,
-                    this.changeFamily.rebaser.invert(branchCommit.changeset),
+                    {
+                        revision: brand(nextRevisionNum--),
+                        change: this.changeFamily.rebaser.invert(changeWithMetadataFromCommit(branchCommit)),
+                    },
                 ),
             commitToRebase.changeset,
         );
 
-        return this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+        const fullyRebasedChange = this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+
+        const shouldRemoveReference = (revision: RevisionTag): boolean =>
+            isTemporaryRevision(revision) || isBranchLocalRevision(branch, revision);
+
+        return this.changeFamily.rebaser.filterReferences(
+            fullyRebasedChange,
+            shouldRemoveReference,
+        )
     }
 
     // TODO: Try to share more logic between this method and `rebaseBranch`
-    private rebaseLocalBranch(trunkChange: TChangeset): TChangeset {
-        const newBranchChanges: TChangeset[] = [];
-        const inverses: TChangeset[] = [];
+    private rebaseLocalBranch(trunkChange: ChangeWithMetadata<TChangeset>): TChangeset {
+        const newBranchChanges: ChangeWithMetadata<TChangeset>[] = [];
+        const inverses: ChangeWithMetadata<TChangeset>[] = [];
 
+        let nextTempRevision = -1;
         for (const localChange of this.localChanges) {
             let change = this.rebaseChange(localChange, inverses);
```

Comment:
I think we want to filter out references to branch changes on the `localChange` (before rebasing it).
My thinking is that...
a. We want to avoid mixing up references to different versions of the same local change.
b. We should not need those references because we now know that they are to changes that will not end up on the trunk.